test/util/stream_utils_unit.js

```diff
@@ -499,4 +499,73 @@ describe('StreamUtils', () => {
       expect(manifest.textStreams[1].id).toBe(2);
     });
   });
+
+  describe('chooseCodecsAndFilterManifest', () => {
+    const addVariant1080Avc1 = (manifest) => {
+      manifest.addVariant(0, (variant) => {
+        variant.bandwidth = 5058558;
+        variant.addAudio(1, (stream) => {
+          stream.bandwidth = 129998;
+        });
+        variant.addVideo(2, (stream) => {
+          stream.bandwidth = 4928560;
+          stream.codecs = 'avc1.640028';
+          stream.size(1920, 1080);
+        });
+      });
+    };
+
+    const addVariant1080Vp9 = (manifest) => {
+      manifest.addVariant(3, (variant) => {
+        variant.bandwidth = 4911000;
+        variant.addAudio(4, (stream) => {
+          stream.bandwidth = 129998;
+        });
+        variant.addVideo(5, (stream) => {
+          stream.bandwidth = 4781002;
+          stream.codecs = 'vp09.00.40.08.00.02.02.02.00';
+          stream.size(1920, 1080);
+        });
+      });
+    };
+
+    const addVariant2160Vp9 = (manifest) => {
+      manifest.addVariant(6, (variant) => {
+        variant.bandwidth = 10850316;
+        variant.addAudio(7, (stream) => {
+          stream.bandwidth = 129998;
+        });
+        variant.addVideo(8, (stream) => {
+          stream.bandwidth = 10784324;
+          stream.codecs = 'vp09.00.40.08.00.02.02.02.00';
+          stream.size(3840, 2160);
+        });
+      });
+    };
+
+    it('chooses variants with different sizes by codecs', () => {
+      manifest = shaka.test.ManifestGenerator.generate((manifest) => {
+        addVariant1080Avc1(manifest);
+        addVariant1080Vp9(manifest);
+        addVariant2160Vp9(manifest);
+      });
+
+      shaka.util.StreamUtils.chooseCodecsAndFilterManifest(manifest);
+
+      expect(manifest.variants.length).toBe(2);
+      expect(manifest.variants[0].id).toBe(3);
```

Comment:
This seems like you're checking for VP9 to have been chosen.  Can you make that more explicit with a check on variants[0].video.codec instead?  As it is now, to understand the test, I have to flip back to the variant definitions to find the ID numbers.